File: src/main/java/io/github/bot/terminal/application/domains/monitoring/rest/dto/MonitoringTaskDTO.java
package io.github.bot.terminal.application.domains.monitoring.rest.dto;

import io.github.bot.terminal.application.domains.common.action.dto.ActionDTO;
import io.github.bot.terminal.application.domains.common.conditions.dto.ConditionDTO;
import io.github.bot.terminal.application.domains.common.notify.dto.NotifyDTO;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@Accessors(chain = true)
public class MonitoringTaskDTO {
    private String id;
    private String name;
    private String state;
    private ActionDTO action;
    private String cron;
    private List<ConditionDTO> conditions;
    private List<NotifyDTO> notifyList;
}

File: src/main/java/io/github/bot/terminal/application/domains/monitoring/rest/dto/MonitoringTaskDTO.kt
package io.github.bot.terminal.application.domains.monitoring.rest.dto

import io.github.bot.terminal.application.domains.common.action.dto.ActionDTO
import io.github.bot.terminal.application.domains.common.conditions.dto.ConditionDTO
import io.github.bot.terminal.application.domains.common.notify.dto.NotifyDTO


data class MonitoringTaskDTO(
        val id: String,
        val name: String,
        val enabled: Boolean,
        val action: ActionDTO,
        val cron: String,
        val conditions: List<ConditionDTO>,
        val notifyList: List<NotifyDTO>
)